plugins/inputs/phpfpm/phpfpm.go

```diff
@@ -277,6 +279,68 @@ func importMetric(r io.Reader, acc telegraf.Accumulator, addr string) (poolStat,
 	return stats, nil
 }
 
+func expandUrls(urls []string) ([]string, error) {
+	addrs := make([]string, 0, len(urls))
+	for _, url := range urls {
+		if isNetworkURL(url) {
+			addrs = append(addrs, url)
+			continue
+		}
+		paths, err := globUnixSocket(url)
+		if err != nil {
+			return nil, err
+		}
+		addrs = append(addrs, paths...)
+	}
+	return addrs, nil
+}
+
+func globUnixSocket(url string) ([]string, error) {
+	pattern, status := unixSocketPaths(url)
+	paths, err := filepath.Glob(pattern)
+	if err != nil {
+		return nil, errors.Wrapf(err, "couldn't read the file path pattern %q", pattern)
+	}
+
+	addrs := make([]string, 0, len(paths))
+
+	if len(paths) == 0 {
+		_, err := os.Stat(pattern)
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("Socket doesn't exist  '%s': %s", pattern, err)
+		}
+		return nil, err
```

Comment:
I did it this way to keep the previous behavior, which is to error out on nonexistent socket paths. Since `filepath.Glob` ignores any errors other than syntax ones, I've used `os.Stat` to get the appropriate error. The special handling of `os.ErrNotExist` is there because it's also done elsewhere in the plugin.

If you think we consider that not matching any sockets is not an error, this code would indeed be a bit cleaner.

Reply:
One example of where this handling is useful is when the unix socket path is in fact *not* a glob, but the file does not exist. In that case, we'd get an empty slice of strings as a result from the glob, with no error, so this extra checking restores the error message one would get before glob support.

Reply:
Okay, let's continue to show an error when the socket doesn't exist.  I think we should then remove the Stat call on L166, which doesn't seem useful anymore and anyway we will report an error afterwards connecting to the socket if it doesn't exist.

Reply:
Looking at the `globpath` module I noticed `Match` actually returns the given parameter when it's a static path, so the latest commit handles this case correctly now. I've also removed the redundant `os.Stat` call.

If the PR is OK now, do you want me to rebase before merging?